h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java

```diff
@@ -21,24 +23,59 @@ public StackedEnsembleMojoModel(String[] columns, String[][] domains, String res
             basePreds = new double[_baseModelNum * _nclasses]; //Proper allocation for multinomial ensemble (class probabilities per base model)
             for(int i = 0; i < _baseModelNum; ++i){
                 for(int j = 0; j < _nclasses; ++j){
-                    basePreds[k] = _baseModels[i].score0(row, basePredsRow)[j+1];
+                    basePreds[k] = _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow)[j + 1];
                     k++;
                 }
             }
         }else if(_nclasses == 2){ //Binomial
             for(int i = 0; i < _baseModelNum; ++i) {
-                _baseModels[i].score0(row, basePredsRow);
+                _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow);
                 basePreds[i] = basePredsRow[2];
             }
         }else{ //Regression
             for(int i = 0; i < _baseModelNum; ++i) { //Regression
-                _baseModels[i].score0(row, basePredsRow);
+                _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow);
                 basePreds[i] = basePredsRow[0];
             }
         }
         _metaLearner.score0(basePreds, preds);
         return preds;
     }
 
+    /**
+     * In stacked ensembles, multiple models may appear. Problem with multiple models present is possibly different
+     * internal order of features. Therefore, the scored row's values must be re-mapped to the internal order of each
```

Comment:
how about taking care of the order when mojo is being persisted? I know it would cost some effort, but perhaps it saves us some time spent on scoring side?

Reply:
A noble goal indeed and we may do it in future, but:
- The performance cost of this is not that high
- It would make older models obsolete and unusable. 

We need to carefully evaluate this. I suggest thinking about it in future, when there are less bad models and we have time to carefully plan the deprecation.

Reply:
true - there are those existing models...
If we take this route, we would have to introduce a flag (maybe model version?) indicating that the model was written with unified list of features. Meaning that in scoring side, we will support both approaches until the end of the deprecation cycle.
Sure, keep me in loop! thanks